app/javascript/components/Appointment.js

```diff
@@ -0,0 +1,155 @@
+                    disabled
+                    type="text"
+                    value={user && user.name}
+                  />
+                </label>
+              </div>
+
+              <div className="form-group mb-2">
+                <label htmlFor="doctor" className="appointment-label w-100">
+                  Select a doctor
+                  <span className="text-danger">*</span>
+                  <br />
+                  <select
+                    className="form-control form-input"
+                    onChange={handleChange}
+                    name="doctor_id"
+                  >
+                    <option value="">Select a doctor</option>
+                    {doctors
+                      && doctors.map((doctor) => (
+                        <option
+                          className="select-input"
+                          key={doctor.id}
+                          value={doctor.id}
+                        >
+                          {doctor.name}
+                        </option>
+                      ))}
+                  </select>
+                </label>
+              </div>
+
+              <div className="form-group mb-2">
+                <span aria-labelledby="date" className="appointment-label">
+                  Date
+                  <span className="text-danger">*</span>
+                  <Calendar
+                    id="date"
+                    value={appointmentDate}
+                    onChange={(e) => setAppointmentDate(e.value)}
+                    showIcon
+                  />
+                </span>
+              </div>
+              <div className="form-group mb-2">
+                <span
+                  aria-labelledby="start_time"
+                  className="appointment-label"
+                >
+                  Start Time
+                  <span className="text-danger">*</span>
+                </span>
+                <Calendar
+                  hourFormat="12"
+                  id="start_time"
+                  value={startTime}
+                  onChange={(e) => setStartTime(e.value)}
+                  timeOnly
+                />
+              </div>
+              <div className="form-group mb-2">
+                <span aria-labelledby="end_time" className="appointment-label">
+                  End Time
+                  <span className="text-danger">*</span>
+                </span>
+                <Calendar
+                  hourFormat="12"
+                  id="calendar-timeonly"
+                  value={endTime}
+                  onChange={(e) => setEndTime(e.value)}
+                  timeOnly
+                />
+              </div>
+              <button
+                type="submit"
+                className="mt-4 btn btn-success-cus d-block w-100 rounded-pill"
+              >
+                Book Appointment
+              </button>
+            </form>
+          </div>
+        </div>
+      </div>
+    </div>
```

Comment:
[OPTIONAL] When you see similar parts being repeated (like same style of div) you can consider simplifying code with loop or similar.